ecs/spotfleet-cluster/spotrequest.tf

```diff
@@ -0,0 +1,2174 @@
+
+echo [general] > /etc/awslogs/awslogs.template
+echo "state_file = /var/lib/awslogs/agent-state" >> /etc/awslogs/awslogs.template
+echo "[/var/log/ecs/ecs-agent.log]" >> /etc/awslogs/awslogs.template
+echo "file = /var/log/ecs/ecs-agent.log*" >> /etc/awslogs/awslogs.template
+echo "log_group_name = ${aws_cloudwatch_log_group.main.name}" >> /etc/awslogs/awslogs.template
+echo "log_stream_name = {instance_id}" >> /etc/awslogs/awslogs.template
+
+cp /etc/awslogs/awscli.template /etc/awslogs/awscli.conf
+cp /etc/awslogs/awslogs.template /etc/awslogs/awslogs.conf
+start ecs
+service awslogs start
+USER_DATA
+  }
+
+  launch_specification {
+    ami                    = "${var.ami}"
+    instance_type          = "t2.medium"
+    subnet_id              = "${var.subnets[0]}"
+    vpc_security_group_ids = ["${aws_security_group.ecs-instances.id}"]
+    weighted_capacity      = 2
+
+    tags = {
+      "Name" = "${var.prefix}-spot-instance"
+    }
+
+    iam_instance_profile = "${aws_iam_instance_profile.ecs.name}"
+
+    user_data = <<USER_DATA
+#!/bin/bash
+yum install -y https://amazon-ssm-${data.aws_region.current.name}.s3.amazonaws.com/latest/linux_amd64/amazon-ssm-agent.rpm
+yum install -y awslogs
+yum install -y aws-cfn-bootstrap
+echo ECS_CLUSTER=${aws_ecs_cluster.main.name} >> /etc/ecs/ecs.config
+echo ECS_LOG_LEVEL=${var.ecs_log_level} >> /etc/ecs/ecs.config
+echo ECS_ENABLE_CONTAINER_METADATA=true >> /etc/ecs/ecs.config
+echo ECS_ENABLE_TASK_IAM_ROLE=true >> /etc/ecs/ecs.config
+echo ECS_ENABLE_TASK_IAM_ROLE_NETWORK_HOST=true >> /etc/ecs/ecs.config
+echo ECS_AVAILABLE_LOGGING_DRIVERS=[\"json-file\",\"awslogs\"] >> /etc/ecs/ecs.config
+
+mkdir /etc/awslogs
+
+echo "[plugins]" > /etc/awslogs/awscli.template
+echo "cwlogs = cwlogs" >> etc/awslogs/awscli.template
+echo "[default]" >> etc/awslogs/awscli.template
+echo "region = ${data.aws_region.current.name}" >> etc/awslogs/awscli.template
+
+echo [general] > /etc/awslogs/awslogs.template
+echo "state_file = /var/lib/awslogs/agent-state" >> /etc/awslogs/awslogs.template
+echo "[/var/log/ecs/ecs-agent.log]" >> /etc/awslogs/awslogs.template
+echo "file = /var/log/ecs/ecs-agent.log*" >> /etc/awslogs/awslogs.template
+echo "log_group_name = ${aws_cloudwatch_log_group.main.name}" >> /etc/awslogs/awslogs.template
+echo "log_stream_name = {instance_id}" >> /etc/awslogs/awslogs.template
+
+cp /etc/awslogs/awscli.template /etc/awslogs/awscli.conf
+cp /etc/awslogs/awslogs.template /etc/awslogs/awslogs.conf
+start ecs
+service awslogs start
+USER_DATA
+  }
+
+  launch_specification {
+    ami                    = "${var.ami}"
+    instance_type          = "t2.medium"
+    subnet_id              = "${var.subnets[1]}"
+    vpc_security_group_ids = ["${aws_security_group.ecs-instances.id}"]
+    weighted_capacity      = 2
+
+    tags = {
+      "Name" = "${var.prefix}-spot-instance"
+    }
+
+    iam_instance_profile = "${aws_iam_instance_profile.ecs.name}"
+
+    user_data = <<USER_DATA
```

Comment:
Use locals, variables, templates, anything instead of copying the same content :)

Reply:
@colincoleman - even if you can't interpolate the entire launch specification, perhaps just the `user_data` can be passed in from a `template_file`? If not, I think we need to either submit a new issue or upvote an existing one 😆

Reply:
I am pretty sure that you can interpolate the static content (user_data)- I will see if I can move that to a cloud config